### PR TITLE
[GOVCMS-4001]: Improved logging around file operations

### PR DESCRIPTION
**Pull request by steveworley**
Lagoon Extras, a place where additional application glue code can be added to provide additional insight to sites running on the platform.

### CODE REVIEW

File: modules/lagoon/lagoon_extras/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# GovCMS Platform
+
+Provides additional tooling and management processes for sites hosted on the govCMS platform. The module provides a number of service decorators to improve our ability to manage the platform.
+
+## Verbose logging
+
+```
+$config['lagoon_extras.settings']['verbose_logging']= getenv('GOVCMS_VERBOSE_LOGGING') ?: FALSE;
```
Comment (srowlands) [inline modules/lagoon/lagoon_extras/README.md]: This is currently missing from settings file, so we have no way to enable via env var at the moment

File: modules/lagoon/lagoon_extras/src/File/FileSystem.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+class FileSystem extends BaseFileSystem {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function unlink($uri, $context = NULL) {
+    $status = parent::unlink($uri, $context);
+
+    if ($status) {
+      $this->logger->info("The file %uri was successfully unlinked", ['%uri' => $uri]);
```
Comment (AlexSkrypnyk) [inline modules/lagoon/lagoon_extras/src/File/FileSystem.php]: should use single quotes

File: modules/lagoon/lagoon_extras/src/File/FileSystem.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      if (\Drupal::config(Form::SETTINGS)->get('verbose_logging')) {
+        $this->logger->info('%backtrace', ['%backtrace' => json_encode(debug_backtrace())]);
```
Comment (AlexSkrypnyk) [inline modules/lagoon/lagoon_extras/src/File/FileSystem.php]: do we want to limit `debug_backtrace()` to use `DEBUG_BACKTRACE_IGNORE_ARGS` to save the memory?

File: modules/lagoon/lagoon_extras/src/Form/SettingsForm.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $this->configFactory->getEditable(static::SETTINGS)
+      ->set('verbose_logging', $form_state->getValue(['verbose_logging']))
+      ->save();
+
+    parent::submitForm($form, $form_state);
+  }
+
```
Comment (AlexSkrypnyk) [inline modules/lagoon/lagoon_extras/src/Form/SettingsForm.php]: empty lines

File: modules/lagoon/lagoon_extras/src/Form/SettingsForm.php [hunk 5]
```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Drupal\lagoon_extras\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
```
Comment (AlexSkrypnyk) [inline modules/lagoon/lagoon_extras/src/Form/SettingsForm.php]: class description is missing

**Review by AlexSkrypnyk — CHANGES_REQUESTED** [review]
@steveworley 
We need to run PHPCS for this module as it currently violates Drupal coding standards.

Also, the module is missing tests.

**Comment by simesy** [comment]
Looking for a little more info in the README on this module. In the past modules that live in govcms8lagoon have only been installed and then uninstalled. So they had no effect on the resulting config state. Having drupal modules in the image repo is an antipattern as far as I'm concerned but I've overlooked it in the past due to the fact the modules get uninstalled afterwards.

This module has routes and settings forms and such, so if it's installed we're relying on operations to uninstall it afterwards. Perhaps a note on how to add this module to config ignore so that it doesn't unintentionally effect config operations.

**Pull request closed without merging**